Rover/sailboat.cpp

```diff
@@ -260,6 +264,57 @@ void Sailboat::get_throttle_and_mainsail_out(float desired_speed, float &throttl
         wingsail_out *= -1.0f;
     }
 
+    //
+    // direct mast rotation control
+    // configure SERVOx_MAX/MIN such that the mast rotates +-90 degrees
+    //
+
+    if (!is_positive(desired_speed)) {
+        // rotating sails can be used to reverse, but not in this version
+        mast_rotation_out = 0.0f;
+    } else {
+        // + is wind over starboard side, - is wind over port side
+        float wind_dir_apparent_signed = degrees(rover.g2.windvane.get_apparent_wind_direction_rad());
+        float wind_dir_apparent_abs = fabsf(wind_dir_apparent_signed);
+        float wind_dir_apparent_sign = wind_dir_apparent_signed / wind_dir_apparent_abs;
```

Comment:
I think its nicer to use the tack flag from wind-vane as we do for wing elevators.

Reply:
I'd be a lot more comfortable using it if it were a function. I'm hesitant to use a stateful variable set somewhere far away, filtered in ways that are not obvious from the call site. Is `AP_WindVane::Sailboat_Tack::TACK_PORT` a statement of fact (wind is coming over the starboard side) or a statement of intent (the AP _wants_ to boat to be on a starboard tack?) Can it get out of sync with the absolute value? What guarantees do I have this will not change in a future PR?

As I've written it, with very local state, the relationship between absolute value and sign is obvious and is likely to stay consistent as the code develops.

https://discord.com/channels/674039678562861068/797946854792757259/830491529654304789

Reply:
Were asking the windvane what tack it thinks were on, its independent of navigation. Once the new wind-vane code is in this will let you filter the tack wind direction differently to the apparent wind direction.

Reply:
I don't want to filter it differently. I want a single, consistent, atomic, signed number for apparent wind, so I can set AoA to it. Remember for a wngsail, I need to know the direction, not just magnitude of the apparent wind. Sailing with sheets is a lot simpler computationally, which is why everyone has been doing it that way since the beginning of sailing until digital computers came along.

I split the sign off for readability. I can rewrite the code so I keep apparent wind signed and replace conditions like
` if (wind_dir_apparent_abs < sail_angle_ideal)` with 
`if ((wind_dir_apparent < sail_angle_ideal) || (wind_dir_apparent > (-1.0f * sail_angle_ideal)))` and there will be no need to argue how to find the tack.